src/s3ql/fs.py

```diff
@@ -932,9 +943,16 @@ async def open(self, id_, flags, ctx):
             and (self.failsafe or self.inodes[id_].locked)):
             raise FUSEError(errno.EPERM)
 
-        if flags & os.O_TRUNC:
-            log.warning('open() with O_TRUNC is not yet supported.')
-            raise FUSEError(errno.ENOTSUP)
+        if (flags & os.O_RDWR or flags & os.O_WRONLY) and flags & os.O_TRUNC:
```

Comment:
So if flags == O_TRUNC | O_RDONLY, we silently ignore the flag? Is this a deliberate choice?

Reply:
The behaviour for `O_TRUNC | O_RDONLY` is undefined:

https://www.man7.org/linux/man-pages/man2/open.2.html#DESCRIPTION
```
       O_TRUNC
              If the file already exists and is a regular file and the
              access mode allows writing (i.e., is O_RDWR or O_WRONLY) it
              will be truncated to length 0.  If the file is a FIFO or ter‐
              minal device file, the O_TRUNC flag is ignored.  Otherwise,
              the effect of O_TRUNC is unspecified.
```

https://www.man7.org/linux/man-pages/man2/open.2.html#NOTES
```
NOTES  
       [...]

       The (undefined) effect of O_RDONLY | O_TRUNC varies among
       implementations.  On many systems the file is actually truncated.

```

> [...]On many systems the file is actually truncated.[...]

That's kinda drastic, I think. Let's throw a EINVAL instead. (for all unspecified combinations)